src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -292,8 +291,8 @@ public async Task StopAsync(CancellationToken cancellationToken = default)
         /// <summary>
         /// Disposes the <see cref="HubConnection"/>.
         /// </summary>
-        /// <returns>A <see cref="Task"/> that represents the asynchronous dispose.</returns>
-        public async Task DisposeAsync()
+        /// <returns>A <see cref="ValueTask"/> that represents the asynchronous dispose.</returns>
+        public async ValueTask DisposeAsync()
```

Comment:
You *could* do this without a breaking change by making a new explicit method:

```csharp
async ValueTask IAsyncDisposable.DisposeAsync()
{
    // Do the things
}

public Task DisposeAsync()
{
    return ((IAsyncDisposable)this).DisposeAsync().AsTask();
}
```

Reply:
Yeah, I did that in 3.0 but we decided to move it to 5.0 and we can make the small break here.